src/macros.rs

```diff
@@ -149,8 +149,7 @@
 ///
 /// In Rust 1.45 we can eliminate this redundancy by allowing casts using `as _` or type ascription
 /// syntax, i.e. `my_int: _` (which is unstable but can be stripped), but this requires modifying
-/// the expression which is not possible as the macros are currently implemented. Casts to `_` are
-/// forbidden for now as they produce rather nasty type errors.
+/// the expression which is not possible as the macros are currently implemented.
```

Comment:
Actually, this whole paragraph is outdated now.

Using `as _` or `: _` simply signals to the macro to not type-check that bind expression and then that syntax is stripped from the expression so as to not trigger type errors (or an unstable syntax feature in the case of the latter, which is called type ascription).

Reply:
you mean like this..? (i just replace the whole paragraph with yours)

Reply:
Yeah, that works.